Create new citation service

jcoyne opened the pull request:
<!-- Closes #ISSUE_NUMBER -->
<!-- 📝 CHANGELOG update? -->

```
search-works(dev)> doc = SolrDocument.find('2472159')
Solr query: get get {qt: nil, ids: "2472159"}
Solr fetch (43.7ms)
=>
#<SolrDocument:20040 _source: {"id" => "2472159", "hashed_id_ssi" => "ebb4709083b1412420969973c0d82e4f", "marc_json_struct" => ["{\"leader\":\"51206cem a2203913 a 450...
search-works(dev)> CitationService.new(doc).export('apa')
=> ["Great Britain, Great Britain Ordnance Survey, &amp; Great Britain Army Royal Engineers. (1958). <i>World 1:500,000</i> [Map]."]
```

cbeer commented:
Here's some odd differences:

Different years:
```
3876636
OLD: Hill, R. (1997). On Beulah Height. HarperCollins.
NEW: Hill, R. (1998). On Beulah Height. HarperCollins.

317712
OLD: Vandier, J. (1952). Manuel d’archéologie égyptienne. A. et J. Picard.
NEW: Vandier, J. (1978). Manuel d'archéologie egyptienne.
```

We're including editors:
```
10689066
OLD: Borghesi, A. (2014). I Sardi e la Resistenza : il contributo dei partigiani di Ardauli alla lotta di Liberazione, 1943-1945 (G. Deiana & V. Urru, Eds.). Iskra.
NEW: Deiana, G., Urru, V., & Borghesi, A. (2014). I Sardi e la Resistenza : il contributo dei partigiani di Ardauli alla lotta di Liberazione, 1943-1945. Iskra.
```

and contributors:
```
2656762
OLD: McKinney, K. C., & United States Office of Educational Research and Improvement. (1987). Parents : here’s how to make school visits work (N. Paulu, Ed.). Office of Educational Research and Improvement, U.S. Dept. of Education.
NEW: McKinney, K., Paulu, N., & United States Office of Educational Research and Improvement. (1987). Parents : here's how to make school visits work. Office of Educational Research and Improvement, U.S. Dept. of Education : For sale by the Supt. of Docs., U.S. G.P.O.
```

Funky punctuation, if nothing else:
```
1150707
OLD: Vanderwilt, J. W., Burbank, W. S., Traver, W. M., & Colorado Mineral Resources Board. (1947). Mineral resources of Colorado. State of Colorado, Mineral Resources Board.
NEW: Colorado, & Geological Survey (U.S.). (1947). Mineral resources of Colorado.
```

Additional publisher information:
```
6827529
OLD: Oldfield, J. R. (2007). ‘Chords of freedom’ : commemoration, ritual and British transatlantic slavery. Manchester University Press.
NEW: Oldfield, J. R. (2007). 'Chords of freedom' : commemoration, ritual and British transatlantic slavery. Manchester University Press ; Distributed in the USA by Palgrave.
```

Funky date and some extra punctuation:
```
9999606
OLD: ESR Inc. (1971). Annual report.
NEW: ESR Inc. (19 C.E.). Annual report. ESR Inc..
```

Duplicate authors:
```
8589252
OLD: Cogen, M. (2008). The comprehensive guide to international law. Die Keure.
NEW: Cogen, M., & Cogen, M. (2008). The comprehensive guide to international law. Die Keure.
```

🤷‍♂️

jcoyne commented:
Here's some odd differences:

Different years:
```
3876636
OLD: Hill, R. (1997). On Beulah Height. HarperCollins.
NEW: Hill, R. (1998). On Beulah Height. HarperCollins.

317712
OLD: Vandier, J. (1952). Manuel d’archéologie égyptienne. A. et J. Picard.
NEW: Vandier, J. (1978). Manuel d'archéologie egyptienne.
```
- [ ] 3876636 has a publication year different from the copyright year. I've fixed this and added a test.
- [ ] 317712 - OCLC doesn't seem to have an end year for this record, but Searchworks does. https://search.worldcat.org/title/Manuel-d%27archeologie-egyptienne/oclc/2528197


We're including editors:
```
10689066
OLD: Borghesi, A. (2014). I Sardi e la Resistenza : il contributo dei partigiani di Ardauli alla lotta di Liberazione, 1943-1945 (G. Deiana & V. Urru, Eds.). Iskra.
NEW: Deiana, G., Urru, V., & Borghesi, A. (2014). I Sardi e la Resistenza : il contributo dei partigiani di Ardauli alla lotta di Liberazione, 1943-1945. Iskra.
```

- [ ] TODO

and contributors:
```
2656762
OLD: McKinney, K. C., & United States Office of Educational Research and Improvement. (1987). Parents : here’s how to make school visits work (N. Paulu, Ed.). Office of Educational Research and Improvement, U.S. Dept. of Education.
NEW: McKinney, K., Paulu, N., & United States Office of Educational Research and Improvement. (1987). Parents : here's how to make school visits work. Office of Educational Research and Improvement, U.S. Dept. of Education : For sale by the Supt. of Docs., U.S. G.P.O.
```
- [ ] TODO

Funky punctuation, if nothing else:
```
1150707
OLD: Vanderwilt, J. W., Burbank, W. S., Traver, W. M., & Colorado Mineral Resources Board. (1947). Mineral resources of Colorado. State of Colorado, Mineral Resources Board.
NEW: Colorado, & Geological Survey (U.S.). (1947). Mineral resources of Colorado.
```
- [ ] TODO

Additional publisher information:
```
6827529
OLD: Oldfield, J. R. (2007). ‘Chords of freedom’ : commemoration, ritual and British transatlantic slavery. Manchester University Press.
NEW: Oldfield, J. R. (2007). 'Chords of freedom' : commemoration, ritual and British transatlantic slavery. Manchester University Press ; Distributed in the USA by Palgrave.
```
- [ ] TODO, I think we need to ignore distribution

Funky date and some extra punctuation:
```
9999606
OLD: ESR Inc. (1971). Annual report.
NEW: ESR Inc. (19 C.E.). Annual report. ESR Inc..
```
- [ ] TODO

Duplicate authors:
```
8589252
OLD: Cogen, M. (2008). The comprehensive guide to international law. Die Keure.
NEW: Cogen, M., & Cogen, M. (2008). The comprehensive guide to international law. Die Keure.
```
- [ ] TODO

jcoyne commented:
Need to add:
- [ ] Place of publication (see 4245187)
- [ ] Records without a 254a (See 4084976)
- [ ] Bad date 666571

cbeer commented:
Can you rebase this? 😬

jcoyne commented:
@cbeer yes, it has been rebased.
